# class CustomTools:
    
#     def _safe_execution(func):
#         def errorHandling(self, *args, **kwargs):
#             try:
#                 return func(self, *args, **kwargs)
#             except Exception as e:
#                 return f"Error while executing {func.__name__}"